### PR TITLE
Functionality #5, added function to check past dates with tests

### PR DESCRIPTION
**Pull request by leeyspaul**
Issue #5 
Added functionality for checking past dates. 

- Tests have been added, two for failing and one pass check, found in `test/index.js`
- New function added under `isDatePast`, utilizes current date from moment `moment().format(currentFormat)` and built in `isBefore()` function 

isBefore documentation: http://momentjs.com/docs/#/query/

screen shot 2018-09-29 at 21 22 11

Let me know if I've messed something up and I'll fix it up as soon as I can! Thanks~

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -116,5 +116,18 @@ module.exports = {
...
+
+  /**
+   * Check if given date is past the current date
+   * @param date
+   * @param currentFormat
+   * @returns {boolean}
+   */
+   isDatePast: function ( date, currentFormat ) {
+     const currentDate = moment().format(currentFormat);
+     let isPast = moment(date).isBefore(currentDate);
```
Comment (arshadkazmi42) [inline index.js]: You can directly return 

`moment(date).isBefore(currentDate);`

as there is no use of using a new variable `isPast`

File: test/index.js [hunk 2]
```diff
@@ -40,3 +40,17 @@ describe("check if date is invalid", function () {
     DateValidator.isValidDate("AA/BB/2017", "DD/MM/YYYY").should.equal(false);
   });
 });
+
+describe("check if date is in the past or not", function() {
+  it("returns false for past check on given date of (2019-09-29, YYYY-MM-DD)", function () {
+    DateValidator.isDatePast("2019-09-29", "YYYY-MM-DD").should.equal(false);
```
Comment (arshadkazmi42) [inline test/index.js]: I think we should use moment for getting future date, so test does not fails even if its `2020`

File: test/index.js [hunk 3]
```diff
@@ -40,3 +40,17 @@ describe("check if date is invalid", function () {
...
+  });
+
+  it("returns false for past check on given date of (2018-10-01, YYYY-MM-DD)", function () {
+    DateValidator.isDatePast("2018-10-01", "YYYY-MM-DD").should.equal(false);
```
Comment (arshadkazmi42) [inline test/index.js]: Same as above, use moment in place of passing the hard coded date

File: test/index.js [hunk 4]
```diff
@@ -40,3 +40,17 @@ describe("check if date is invalid", function () {
...
+  });
+
+  it("returns true for past check on given date of (2004-10-01, YYYY-MM-DD)", function () {
+    DateValidator.isDatePast("2004-10-01", "YYYY-MM-DD").should.equal(true);
```
Comment (arshadkazmi42) [inline test/index.js]: same here also.

**Comment by arshadkazmi42** [comment]
@leeyspaul I have suggested few changes.
Also can you tag the issue number in PR description

File: index.js (hunk 1 above)
Comment (leeyspaul) [reply index.js]: Ok just did this one.

File: test/index.js (hunk 2 above)
Comment (leeyspaul) [reply test/index.js]: So I'm thinking of the `moment().add()` functionality. Is that what you're thinking of?

Comment (arshadkazmi42) [reply test/index.js]: yes. thats what I had in my mind

**Comment by leeyspaul** [comment]
Ok I fixed the `isDatePast` function as requested and then utilized `moment().add(...)` and `.subtract(...)` functionality for tests. Although the number is still hardcoded into add and subtract. Wasn't sure if this was wanted, do let me know!~

File: test/index.js [hunk 5]
```diff
@@ -40,3 +41,14 @@ describe("check if date is invalid", function () {
...
   });
 });
+
+describe("check if date is in the past or not", function() {
+  it("returns false for past check on a given random date in the future", function () {
+    DateValidator.isDatePast(moment().add(5, 'days'), "YYYY-MM-DD").should.equal(false);
+  });
+
+  it("returns true for past check on given date random date in the past", function () {
+    console.log(Math.floor(Math.random()) * 100);
```
Comment (arshadkazmi42) [inline test/index.js]: Do we need this console log here?
If its not required you can remove it, we would like to keep the code cleaner.

**Comment by arshadkazmi42** [comment]
@leeyspaul This is awsome, just a small change and we are good to go.

File: test/index.js (hunk 5 above)
Comment (leeyspaul) [reply test/index.js]: Er sorry. I totally forgot that haha

**Comment by arshadkazmi42** [comment]
@leeyspaul Cool. waiting for the checks to complete. Will merge it as soon as tests are completed.

**Pull request merged**